fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -4160,7 +4164,42 @@ public void alterTableProperties(Database db, OlapTable table, Map<String, Strin
                                 ImmutableMap.of(key, propertiesToPersist.get(key)));
                 GlobalStateMgr.getCurrentState().getEditLog().logAlterTableProperties(info);
             }
+            if (propertiesToPersist.containsKey(PropertyAnalyzer.PROPERTIES_DATACACHE_ENABLE)) {
+                String dataCacheEnable = propertiesToPersist.get(PropertyAnalyzer.PROPERTIES_DATACACHE_ENABLE);
+                boolean isEnable = Boolean.parseBoolean(dataCacheEnable);
 
+                if (table instanceof LakeTable) {
+                    LakeTable lakeTable = (LakeTable) table;
+                    GlobalStateMgr.getCurrentState().getStarOSAgent().updateShardGroup(lakeTable, isEnable);
+                } else {
+                    throw new DdlException("Only support alter datacache.enable for lake tables");
```

Comment:
there is no lake tables to end user, check `ErrorReportException`, if we can add new error code or reuse similar error code.

Reply:
Discuss it with you face to face tomorrow